spinn_utilities/make_tools/file_convertor.py

```diff
@@ -0,0 +1,453 @@
+        raise Exception("Unexpected line {} at {} in {}".format(
+            self._text, self._line_num, self._src))
+
+    def _process_line_in_comment(self, dest_f):
+        if "*/" in self._text:
+            stripped = self._text.strip()
+            match = END_COMMENT_REGEX.search(stripped)
+            if match.end(0) == len(stripped):
+                # Ok Comment until end of line
+                dest_f.write(self._text)
+                self._status = NORMAL_CODE
+                return True
+            return False  # Stuff after comment so check by character
+        # Whole line in comment without end
+        dest_f.write(self._text)
+        return True
+
+    def _process_line_comment_start(self, dest_f):
+        """ Processes a line known assumed to contain a /* but not know where
+
+        There is aslo the assumption that the start status is not COMMENT
+
+        :param dest_f:
+        :return:
+        """
+        stripped = self._text.strip()
+        if stripped.startswith("/*"):
+            self._previous_status = self._status
+            self._status = COMMENT
+            # Comment start so now check for comment end
+            return self._process_line(dest_f)
+        # Stuff before comment so check by char
+        return False  # More than one possible end so check by char
+
+    def _process_line_in_log(self, dest_f):
+        stripped = self._text.strip()
+        if stripped.startswith("//"):
+            # Just a comment line so write and move on
+            dest_f.write(self._text)
+            return True
+
+        match = LOG_END_REGEX.search(stripped)
+        if not match:
+            if stripped[-1:] == ")":
+                # possible start of end
+                self._status = IN_LOG_CLOSE_BRACKET
+            self._log_full += stripped
+            self._log_lines += 1
+            return True
+        if match.end(0) < len(stripped):
+            # Stuff after the log_end so check by char
+            return False
+
+        self._log_lines += 1
+        self._log_full += stripped
+        self._write_log_method(dest_f)
+        self._status = NORMAL_CODE
+        return True
+
+    def _process_line_in_log_close_bracket(self, dest_f):
+        stripped = self._text.strip()
+        if stripped[0] == ";":
+            if stripped == ";":
+                self._log_full += (";")
+                self._log_lines += 1
+                self._write_log_method(dest_f)
+                self._status = NORMAL_CODE
+                return True
+            else:
+                return False
+
+        elif stripped.startswith("//"):
+            # Just a comment line so write and move on
+            dest_f.write(self._text)
+            return True
+
+        else:
+            # so not a closing bracket so set status back
+            self._status = IN_LOG
+            return self._process_line_in_log(dest_f)
+
+    def _process_line_normal_code(self, dest_f):
+        # Fast check
```

Comment:
dont think you need the faster one. when your doing both and the second covers the first.

Reply:
Nuked faster method as wasted more time remembering it then it will ever save